### PR TITLE
Refactor `MacOS::languages`.

### PR DESCRIPTION
**Pull request by reitermarkus**

### CODE REVIEW

File: Library/Homebrew/os/mac.rb [hunk 1]
```diff
@@ -47,19 +47,15 @@ def cat
     end
 
     def languages
-      return @languages unless @languages.nil?
+      return @languages if defined?(@languages)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/os/mac.rb]: The `@languages ||= begin` pattern I think is generally nicest here but don't feel super strongly.

File: Library/Homebrew/os/mac.rb [hunk 2]
```diff
@@ -47,19 +47,15 @@ def cat
...
-      if ENV["HOMEBREW_LANGUAGES"]
-        @languages = ENV["HOMEBREW_LANGUAGES"].split(",") + @languages
-      end
-
-      if ARGV.value("language")
-        @languages = ARGV.value("language").split(",") + @languages
-      end
-
-      @languages = @languages.uniq
+      @languages = [
```
Comment (MikeMcQuaid) [inline Library/Homebrew/os/mac.rb]: `@languages.prepend` or something feels nicer?

Comment (reitermarkus) [reply Library/Homebrew/os/mac.rb]: Doesn't really feel nicer to me.

Comment (MikeMcQuaid) [reply Library/Homebrew/os/mac.rb]: It feels a little weird to be doing a splat on `@languages` inside it's own assignment.

Comment (reitermarkus) [reply Library/Homebrew/os/mac.rb]: It's now all in one assignment.

**Pull request merged**